lib/BloqadeQMC/src/ising/Rydberg.jl

```diff
@@ -1,23 +1,36 @@
 using Base.Iterators
-
+using BloqadeLattices: rydberg_interaction_matrix, AtomList
 
 abstract type AbstractRydberg{O <: AbstractOperatorSampler} <: AbstractLTFIM{O} end
 
-struct Rydberg{O,M <: UpperTriangular{Float64},UΩ <: AbstractVector{Float64}, Uδ <: AbstractVector{Float64}, L <: Lattice} <: AbstractRydberg{O}
+# struct Rydberg{O,M <: UpperTriangular{Float64},UΩ <: AbstractVector{Float64}, Uδ <: AbstractVector{Float64}, L <: Lattice} <: AbstractRydberg{O}
+#     op_sampler::O
+#     V::M
+#     Ω::UΩ
+#     δ::Uδ
+#     lattice::L
+#     energy_shift::Float64
+# end
+# nspins(H::Rydberg) = nspins(H.lattice)
+
+###################
+
+struct Rydberg{O,M <: Matrix{Float64},UΩ <: AbstractVector{Float64}, Uδ <: AbstractVector{Float64}, A} <: AbstractRydberg{O}
```

Comment:
```suggestion
struct Rydberg{O,M <: AbstractMatrix{Float64},UΩ <: AbstractVector{Float64}, Uδ <: AbstractVector{Float64}, A} <: AbstractRydberg{O}
```

In this case we can accept any AbstractMatrix as a type.